pkg/policy/api/utils.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Len returns the total number of rules inside `L7Rules`.
+func (rules *L7Rules) Len() int {
+	return len(rules.HTTP) + len(rules.Kafka)
+}
+
+// Exists returns true if the HTTP rule already exists in the list of rules
+func (h *PortRuleHTTP) Exists(rules L7Rules) bool {
```

Comment:
Should this operate on a copy of L7Rules, or just the pointer?

Reply:
A copy is fine here, the rule data is only read.